src/evaluators/JSXElement.js

```diff
@@ -0,0 +1,298 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+/* @flow */
+
+import type { Realm } from "../realm.js";
+import type { LexicalEnvironment } from "../environment.js";
+import type { Reference } from "../environment.js";
+import type { BabelNodeJSXElement, BabelNodeJSXIdentifier } from "babel-types";
+import { StringValue, ConcreteValue, Value, NumberValue } from "../values/index.js";
+import {
+  GetValue,
+  ToString,
+  ResolveBinding,
+  ArrayCreate,
+  CreateDataPropertyOrThrow,
+  ObjectCreate,
+  Set,
+} from "../methods/index.js";
+import invariant from "../invariant.js";
+import * as t from "babel-types";
+
+let RESERVED_PROPS = {
+  key: true,
+  ref: true,
+  __self: true,
+  __source: true,
+};
+
+let reactElementSymbol;
+let reactElementSymbolKey = "react.element";
+
+// takan from Babel
```

Comment:
Spelling

Also, I would prefer returning args to passing in an array to be mutated as all the callsites seem to pass empty arrays.

Reply:
All the evaluation code was pretty much copy and pasted from the React compiler – I will go through and add more Flow annotations. Thanks :)